scripts/management/MgmtDB.py

```diff
@@ -276,18 +276,24 @@ def get_runnable_tasks(
 
         return runnable_tasks
 
-    def is_task_complete(self, task):
-        process, run_name, status = task
+
+    def num_task_complete(self, task, like=False):
+        process, run_name = task
```

Comment:
task is a 3-tuple, so should throw an unpacking exception?

Reply:
task doesn't need to be a 3-tuple, so I removed "status", which should be a constant "complete"

Reply:
I'd prefer if we didn't use two different tuple structures for a `task` I think using the thruple paradigm everywhere is a better solution than using a two-ple just for completion checking. 

Ideally, the tuple would be refactored into something with an implicit structure but that's well beyond the scope of this.

Reply:
The only place `is_task_complete` called is auto_submit.py - line 526, and originally it had the third element in a list (not even a tuple, which doesn't matter) - the third element was not necessary.

```
                if mgmt_db.is_task_complete(
                    [
                        const.ProcessType.clean_up.value,
                        run_name,
                        const.Status.completed.str_value,
                    ]
                ):
``` 

I am not the inventor of two different tuple structures for a `task`.  See line 295 in MgmtDB.py

```
    def _check_dependancy_met(self, task, logger=get_basic_logger()):
        """Checks if all dependencies for the specified are met"""
        process, run_name = task
        process = Process(process)
```

Reply:
@jasonmotha  It looks like in most places task is two-ple, the thruple seems to be less common. It should probably be a NamedTuple, but that's beyond this PR as you mention.